src/librustdoc/html/render/mod.rs

```diff
@@ -1370,13 +1385,49 @@ impl Setting {
                 if *default_value { " checked" } else { "" },
                 description,
             ),
+            Setting::Select {
+                ref js_data_name,
+                ref description,
+                ref default_value,
+                ref options,
+            } => format!(
+                "<div class='setting-line'>\
+                     <div>{}</div>\
+                     <label class='select-wrapper'>\
+                         <select id='{}' autocomplete='off'>{}</select>\
+                         <img src='./down-arrow.svg' alt='Select item'>\
+                     </label>\
+                 </div>",
+                description,
+                js_data_name,
+                options
+                    .iter()
+                    .map(|opt| format!(
+                        "<option value=\"{}\" {}>{}</option>",
+                        opt.0,
+                        if &opt.0 == *default_value { "selected" } else { "" },
+                        opt.1,
+                    ))
+                    .collect::<String>(),
+            ),
         }
     }
 }
 
 impl From<(&'static str, &'static str, bool)> for Setting {
     fn from(values: (&'static str, &'static str, bool)) -> Setting {
-        Setting::Entry { js_data_name: values.0, description: values.1, default_value: values.2 }
+        Setting::Toggle { js_data_name: values.0, description: values.1, default_value: values.2 }
+    }
+}
+
+impl From<(&'static str, &'static str, &'static str, Vec<(String, String)>)> for Setting {
+    fn from(values: (&'static str, &'static str, &'static str, Vec<(String, String)>)) -> Setting {
+        Setting::Select {
+            js_data_name: values.0,
+            description: values.1,
+            default_value: values.2,
+            options: values.3,
+        }
```

Comment:
This looks pretty confusing, I would just construct it explicitly with struct initialization syntax.

Reply:
Good point, I tried to follow what was already here but since it was just on/off toggles it wasn't as complex. I'll change that tomorrow!